registry/registry.go

```diff
@@ -133,9 +114,11 @@ func ReadCertsDirectory(tlsConfig *tls.Config, directory string) error {
 // DockerHeaders returns request modifiers that ensure requests have
 // the User-Agent header set to dockerUserAgent and that metaHeaders
 // are added.
-func DockerHeaders(metaHeaders http.Header) []transport.RequestModifier {
+func DockerHeaders(userAgent string, metaHeaders http.Header) []transport.RequestModifier {
 	modifiers := []transport.RequestModifier{
-		transport.NewHeaderRequestModifier(http.Header{"User-Agent": []string{dockerUserAgent}}),
+		transport.NewHeaderRequestModifier(http.Header{
+			"User-Agent": []string{userAgent},
```

Comment:
Ok, thanks for elaborating, I didn't get that from the previous comment.

I can have it skip adding the userAgent if it's blank, although an error still feels more correct to me. I believe the primary responsibility of this function was to ensure that we sent a user agent with the relevant docker version information.

Changing it to not error will make it very easy for a new call site to just omit the user agent.

But if that is not a concern, I can make it skip setting a userAgent when the string empty is empty.